### PR TITLE
Bugfix payment rejected if there was a payment after one of the acceptances

### PR DESCRIPTION
**Pull request by Jakub89**
Resolves: #983

### CODE REVIEW

File: concent_api/core/tests/test_payments_bankster.py [hunk 1]
```diff
@@ -411,38 +401,52 @@ def test_that_settle_overdue_acceptances_should_return_claim_deposit_with_amount
...
+                        provider_ethereum_address=task_to_compute.provider_ethereum_address,
+                        acceptances=subtask_results_accepted_list,
+                        requestor_public_key=hex_to_bytes_convert(task_to_compute.requestor_public_key),
+                    )
+
+        get_deposit_value_mock.assert_called_once()
         get_list_of_payments_mock.assert_called()
 
+        self.assertIsNotNone(claim_against_requestor.tx_hash)
+
```
Comment (rwrzesien) [inline concent_api/core/tests/test_payments_bankster.py]: Please remove blank line.

File: concent_api/core/tests/test_payments_bankster.py [hunk 2]
```diff
@@ -411,38 +401,52 @@ def test_that_settle_overdue_acceptances_should_return_claim_deposit_with_amount
 
         self.assertEqual(claim_against_requestor.amount, subtask_cost)
 
-    def test_that_settle_overdue_acceptances_should_raise_exception_if_any_transaction_from_acceptances_list_is_before_youngest_transaction_timestamp(self):
+    def test_that_settle_overdue_acceptances_should_return_claim_deposit_with_amount_paid_when_requesting_payment_for_many_subtasks(self):
```
Comment (rwrzesien) [inline concent_api/core/tests/test_payments_bankster.py]: Those are not `many_subtasks`, but exactly two.

File: concent_api/core/tests/test_payments_bankster.py [hunk 3]
```diff
@@ -411,38 +401,52 @@ def test_that_settle_overdue_acceptances_should_return_claim_deposit_with_amount
...
+                        acceptances=subtask_results_accepted_list,
+                        requestor_public_key=hex_to_bytes_convert(task_to_compute.requestor_public_key),
+                    )
+
+        get_deposit_value_mock.assert_called_once()
         get_list_of_payments_mock.assert_called()
 
+        self.assertIsNotNone(claim_against_requestor.tx_hash)
+
+        self.assertEqual(claim_against_requestor.amount, 2000)
```
Comment (rwrzesien) [inline concent_api/core/tests/test_payments_bankster.py]: Add comment how this value was calculated.

File: concent_api/core/tests/test_payments_bankster.py [hunk 4]
```diff
@@ -263,7 +262,7 @@ def test_that_settle_overdue_acceptances_should_return_none_if_subtask_costs_whe
         subtask_results_accepted_list = [
             self._get_deserialized_subtask_results_accepted(
                 report_computed_task=self._get_deserialized_report_computed_task(
-                    timestamp="2018-02-05 10:00:05",
+                    timestamp="2018-02-04 10:00:05",
```
[the remark this answers is not in the document]
Comment (Jakub89) [reply concent_api/core/tests/test_payments_bankster.py]: I've change this timestamp in other tests, so they would be more realistic and reflect blueprint. Here I just wanted to unify them all. I can revert that change if You want?

File: concent_api/core/tests/test_payments_bankster.py (hunk 2 above)
Comment (Jakub89) [reply concent_api/core/tests/test_payments_bankster.py]: True xD can I change it at least to "multiple"?

File: concent_api/core/tests/test_payments_bankster.py (hunk 3 above)
Comment (Jakub89) [reply concent_api/core/tests/test_payments_bankster.py]: Done

File: concent_api/core/tests/test_payments_bankster.py (hunk 1 above)
Comment (Jakub89) [reply concent_api/core/tests/test_payments_bankster.py]: I've added that comment that You requested below, so I think it's justify right now.

**Pull request merged**